src/buffered-process.coffee

```diff
@@ -159,6 +160,22 @@ class BufferedProcess
           process.kill(pid)
       @killProcess()
 
+  # Kill all child processes spawned by a spawned child process
+  #
+  # Uses ps-tree, so it works on *nix systems only
+  killOnNix: ->
+    return unless @process?
+
+    parentPid = @process.pid
+    psTree parentPid, (err, children) ->
+        pidsToKill = children.map (p) -> p.PID
+        for pid in pidsToKill
+          try
+            process.kill(pid)
+        return
+
+    @killProcess()
```

Comment:
Remember to change the outermost arrow to a fat arrow to capture the scope, otherwise `@killProcess()` is going to target the wrong scope. 

Thanks for sticking with me on this!